python/tests/unit/arcticdb/version_store/test_column_type_changes.py

```diff
@@ -90,4 +93,32 @@ def test_changing_fixed_string_width(version_store_factory, dynamic_schema, wide
 
     expected_update = pd.DataFrame({"col": ["aa", "d" * (1 if wider_strings_first else 3), "cc"]}, index=pd.date_range("2024-01-01", periods=3))
     received_update = lib.read(sym_update).data
-    assert_frame_equal(expected_update, received_update)
\ No newline at end of file
+    assert_frame_equal(expected_update, received_update)
+
+def test_type_promotion_stored_in_index_key(version_store_factory):
```

Comment:
NIT: You can use the `lmdb_version_store_dynamic_schema` to cover for both encoding versions and avoid the call to  `version_store_factory(dynamic_schema=True)`. I'm not sure if you can call `library_tool` on it, though...